### PR TITLE
Partially map a weird long set

### PR DESCRIPTION
**Pull request by liach**
Names open to improvement

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/util/collection/HashBasedLongSet.mapping [hunk 1]
```diff
@@ -0,0 +1,37 @@
+CLASS net/minecraft/class_6136 net/minecraft/util/collection/HashBasedLongSet
+	FIELD field_31715 map Lnet/minecraft/class_6136$class_6137;
+	METHOD <init> (IF)V
+		ARG 1 expected
+		ARG 2 f
+	CLASS class_6137 HashBasedMap
+		FIELD field_31720 SECOND_CHARACTERISTICS_OFFSET I
+		FIELD field_31721 FIRST_CHARACTERISITCS_OFFSET I
```
Comment (haykam821) [inline mappings/net/minecraft/util/collection/HashBasedLongSet.mapping]:
```suggestion
		FIELD field_31721 FIRST_CHARACTERISTICS_OFFSET I
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/collection/HashBasedLongSet.mapping [hunk 2]
```diff
@@ -0,0 +1,37 @@
+CLASS net/minecraft/class_6136 net/minecraft/util/collection/HashBasedLongSet
+	FIELD field_31715 map Lnet/minecraft/class_6136$class_6137;
+	METHOD <init> (IF)V
+		ARG 1 expected
+		ARG 2 f
```
Comment (YanisBft) [inline mappings/net/minecraft/util/collection/HashBasedLongSet.mapping]:
```suggestion
		ARG 2 loadFactor
```

File: mappings/net/minecraft/util/collection/HashBasedLongSet.mapping [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	CLASS class_6137 HashBasedMap
+		FIELD field_31720 SECOND_CHARACTERISTICS_OFFSET I
+		FIELD field_31721 FIRST_CHARACTERISTICS_OFFSET I
+		FIELD field_31722 FULL_CHARACTERISTICS_BITS J
+		FIELD field_31723 lastPos I
+		FIELD field_31724 lastCharacteristicLess J
+		FIELD field_31725 capacity I
+		METHOD <init> (IF)V
+			ARG 1 expected
+			ARG 2 f
```
Comment (YanisBft) [inline mappings/net/minecraft/util/collection/HashBasedLongSet.mapping]:
```suggestion
			ARG 2 loadFactor
```

**Comment by Sollace** [comment]
I may have unintentionally mapped this class already. Heh. Feel free to use either one or pick the names you like to improve this PR. #2323

**Comment by Earthcomputer** [comment]
At first glance, this class looks like it's supposed to contain values from `BlockPos.toLong()`. The `MathHelper.log2(60000000)` is very suspicious.

**Comment by Sollace** [comment]
@Earthcomputer I believe it's related to LevelPropagator, as that's the only place I can find which uses this type's parent class.

```
            this.pendingIdUpdatesByLevel[l] = new LongLinkedOpenHashSet(expectedLevelSize, 0.5f){
                @Override
                protected void rehash(int newN) {
                    if (newN > j) {
                        super.rehash(newN);
                    }
                }
            };
```
Likely implemented for use here, but left out for some unknown reason.

**Comment by liach** [comment]
Closing in favor of #2323

**Pull request closed without merging**